compiler/scenario-service/client/src/DA/Daml/LF/PrettyScenario.hs

```diff
@@ -810,15 +808,36 @@ renderValue world name = \case
         renderField (Field label mbValue) =
             renderValue world (name ++ [TL.toStrict label]) (fromJust mbValue)
 
+templateConName :: LF.World -> Identifier -> Maybe (LF.Qualified LF.TypeConName)
+templateConName world (Identifier mbPkgId (TL.toStrict -> qualName)) = do
+  (modName, defName, mod0) <- lookupModuleFromQualifiedName world mbPkgId qualName
+  tpl <- NM.lookup (Tagged [defName]) (LF.moduleTemplates mod0)
+  let pkgRef = case mbPkgId of
+                Just (PackageIdentifier (Just (PackageIdentifierSumPackageId pkgId))) -> LF.PRImport $ Tagged $ TL.toStrict pkgId
+                _ -> LF.PRSelf
+  return (LF.Qualified pkgRef modName (LF.tplTypeCon tpl))
+
+
+renderHeader :: LF.World -> Identifier -> [T.Text]
+renderHeader world identifier = case templateConName world identifier of 
+  Just typeConName -> 
+    case (LF.lookupDataType typeConName world) of
+      Right (LF.DefDataType {..} ) -> map unTagged (map fst dataParams)
+      Left  _ -> []
+  Nothing -> []
+
+
 renderRow :: LF.World -> S.Set T.Text -> NodeInfo -> (H.Html, H.Html)
 renderRow world parties NodeInfo{..} =
-    let (ths, tds) = renderValue world [] niValue
+    let (_, tds) = renderValue world [] niValue
+        ths = renderHeader world niTemplateId
         header = H.tr $ mconcat
             [ foldMap (H.th . (H.div H.! A.class_ "observer") . H.text) parties
             , H.th "id"
             , H.th "status"
-            , ths
+            , foldMap (H.th . H.text) ths
             ]
+        _test = renderHeader world niTemplateId
```

Comment:
Is this line redundant?

Reply:
Yes. Trying to debug the rendering part still.